File: BST/java/Node.java
public class Node {
    int value;
    Node left;
    Node right;

    public Node(int value){
        this.value = value;
        this.left = null;
        this.right = null;
    }

    public boolean add(int value) {
        if(value<this.value){
            if(this.left==null){
                this.left = new Node(value);
                return true;
            }
            return this.left.add(value);
        }

        if(value>this.value){
            if(this.right==null){
                this.right = new Node(value);
                return true;
            }
            return this.right.add(value);
        }
        return false;
    }

    public boolean contain(int value) {
        if(value<this.value){
            if(this.left==null){
                return false;
            }
            return this.left.contain(value);
        }

        if(value>this.value){
            if(this.right==null){
                return false;
            }
            return this.right.contain(value);
        }
        return true;
    }

    public int length(){
        int a = (this.left == null) ? 0 : this.left.length();
        int b = (this.right == null) ? 0 : this.right.length();
        return a + b + 1;

    }
    public int height(){
        int a = (this.left == null) ? 1 : this.left.height() + 1;
        int b = (this.right == null) ? 1 : this.right.height() + 1;
        return (a>b)? a : b;
    }
    
}

File: BST/julia/node.jl
mutable struct Node
    value::Int 
    left::Union{Node, Nothing}
    right::Union{Node, Nothing}
end

function add(self::Node, value::Int)
    if value < self.value
        if isnothing(self.left)
            self.left = Node(value, nothing, nothing)
            return true
        end
        return add(self.left, value)
    elseif value > self.value
        if isnothing(self.right)
            self.right = Node(value, nothing, nothing)
            return true
        end
        return add(self.right, value)
    end

    return false
end

function contain(self::Node, value::Int)
    if value < self.value
        if isnothing(self.left)
            return false
        end
        return contain(self.left, value)
    elseif value > self.value
        if isnothing(self.right)
            return false
        end
        return contain(self.right, value)
    end

    return true
end

function length(self::Node)
    a = isnothing(self.left) ? 0 : length(self.left)
    b = isnothing(self.right) ? 0 : length(self.right)
    return a + b + 1
end

function height(self::Node)
    a = isnothing(self.left) ? 1 : height(self.left) + 1
    b = isnothing(self.right) ? 1 : height(self.right) + 1
    return max(a, b)
end
